porcupine/plugins/filemanager.py

```diff
@@ -122,6 +157,28 @@ def rename(old_path: Path) -> None:
         tab.path = new_path / tab.path.relative_to(old_path)
 
 
+def paste(new_path: Path) -> None:
+    assert copy_path is not None
+    paste_here(new_path.parent)
+
+
+def paste_here(new_path: Path) -> None:
+    assert copy_path is not None
+    new_file_path = new_path / copy_path.name
+
+    if new_file_path.exists():
+        new_file_path = ask_name_for_renaming(new_file_path, is_paste=True)  # rename-overwrite file conflict
+
+    if new_file_path is not None and not copy_path == new_file_path:
```

Comment:
This is a bit hard to read, because the `None` check is not near the code that can set `new_file_path` to `None`. How about this?

```suggestion
    if new_file_path.exists():
        new_file_path = ask_name_for_renaming(new_file_path, is_paste=True)  # rename-overwrite file conflict
        if new_file_path is None:
            return

    if copy_path != new_file_path:
```